sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -134,6 +148,45 @@ public class DataLakeSasBuilder
         /// </summary>
         public string ContentType { get; set; }
 
+        /// <summary>
+        /// Optional. Beginning in version 2020-02-10, this value will be used for
+        /// the AAD Object ID of a user authorized by the owner of the
+        /// User Delegation Key to perform the action granted by the SAS.
+        /// The Azure Storage service will ensure that the owner of the
+        /// user delegation key has the required permissions before granting access.
+        /// If <see cref="_performPosixCheck"/> is enabled, the Azure Storage Service
+        /// will perform an additional POSIX ACL check to determine if the user is
+        /// authorized to perform the requested operation. If not enabled no additional
+        /// permission check for the user specified in this value will be performed.
+        /// This cannot be used in conjuction with. This is only used with generating
+        /// User Delegation SAS.
+        /// </summary>
+        private string _agentObjectId { get; set; }
```

Comment:
Do we want this settable here and in the setter below?

Reply:
I set the variable as private, so it should only be accessible only in the class.